elastic/kibana
# [Index Management] Fix encoding issue on index details page

## Summary
Fixes https://github.com/elastic/kibana/issues/166100

This PR adds a workaround fix for the new index details page when opening for index names with special characters, for example `test_index%`. Because of how encoding/decoding works, we can't use the index name as a part of the url like `/indices/${indexName}` (see for more details). Instead we have to pass the index name in a query parameter like `/indices/index_details?indexName=${indexName}. The downside of this workaround is that the url semantics doesn't reflect that the index name is mandatory for the page to work. Once https://github.com/elastic/kibana/issues/132600 is resolved, we should revert this workaround and use the index name as a url segment again. 

Note for reviewers: The jest tests for this fix are part of https://github.com/elastic/kibana/pull/165705

### How to test
1. Add `xpack.index_management.dev.enableIndexDetailsPage: true` to the file `config/kibana.dev.yml` to enable the new index details page
2. Navigate to Index Management and use the "create index" button 
3. Type a name with special characters, for example `test%`
4. Click the new index name in the list and check that the details page and all tabs work
5. Also reload the page completely and check that the page still loads correctly
